tasks/LaserScanTask.cpp

```diff
@@ -71,39 +74,74 @@ void LaserScanTask::cleanupHook()
     LaserScanTaskBase::cleanupHook();
 }
 
-void LaserScanTask::readInput(ConstLaserScanStampedPtr & laserScanMSG)
+void LaserScanTask::readInput(ConstLaserScanStampedPtr& laserScanMSG)
 {
     if (state() != RUNNING) {
         return;
     }
 
     lock_guard<mutex> readGuard(readMutex);
-
+    unsigned int scan_size = laserScanMSG->scan().ranges_size();
     double range_min = laserScanMSG->scan().range_min();
     double range_max = laserScanMSG->scan().range_max();
-
-    scan.time = getCurrentTime(laserScanMSG->time());
-    scan.minRange = meters_to_milimeters(range_min);
-    scan.maxRange = meters_to_milimeters(range_max);
-    scan.angular_resolution = laserScanMSG->scan().angle_step();
-    scan.start_angle = laserScanMSG->scan().angle_min();
-
-    unsigned int scan_size = laserScanMSG->scan().ranges_size();
-    scan.ranges.resize(scan_size);
-
-    for (unsigned int i = 0; i < scan_size; ++i) {
-        double range = laserScanMSG->scan().ranges(i);
-
-        if (range >= range_max) {
-            scan.ranges[i] = base::samples::TOO_FAR;
+    if (laserScanMSG->scan().has_vertical_count()) {
+        if (m_depth_map.vertical_size != laserScanMSG->scan().vertical_count()) {
+            m_depth_map.vertical_size = laserScanMSG->scan().vertical_count();
+        }
+        if (m_depth_map.horizontal_size != laserScanMSG->scan().count()) {
+            m_depth_map.horizontal_size = laserScanMSG->scan().count();
+        }
+        if (m_depth_map.vertical_interval[0] !=
+                laserScanMSG->scan().vertical_angle_max() ||
+            m_depth_map.vertical_interval[1] !=
+                laserScanMSG->scan().vertical_angle_min()) {
+            m_depth_map.vertical_interval[0] = laserScanMSG->scan().vertical_angle_max();
+            m_depth_map.vertical_interval[1] = laserScanMSG->scan().vertical_angle_min();
         }
-        else if (range <= range_min) {
-            scan.ranges[i] = base::samples::TOO_NEAR;
+        if (m_depth_map.horizontal_interval[0] != laserScanMSG->scan().angle_min() ||
+            m_depth_map.horizontal_interval[1] != laserScanMSG->scan().angle_max()) {
+            m_depth_map.horizontal_interval[0] = laserScanMSG->scan().angle_min();
+            m_depth_map.horizontal_interval[1] = laserScanMSG->scan().angle_max();
         }
-        else {
-            scan.ranges[i] = meters_to_milimeters(range);
+        if (m_depth_map.distances.size() != scan_size) {
+            m_depth_map.distances.resize(scan_size);
         }
+        m_depth_map.time = getCurrentTime(laserScanMSG->time());
+        m_depth_map.timestamps[0] = getCurrentTime(laserScanMSG->time());
+        for (unsigned int i = 0; i < scan_size; ++i) {
+            double range = laserScanMSG->scan().ranges(i);
+            if (range >= range_max || range <= range_min) {
+                m_depth_map.distances[i] = base::unknown<double>();
+            }
```

Comment:
Is that possible ? I would expect the message to already clamp at min/max since it is part of the definition of the ray. Do you know what gazebo does ?

Reply:
Gazebo already reject the samples out of range.
removing that